core/src/main/java/com/linecorp/armeria/internal/common/RequestContextUtil.java

```diff
@@ -48,6 +58,66 @@
     private static final Set<Thread> REPORTED_THREADS =
             Collections.newSetFromMap(new MapMaker().weakKeys().makeMap());
 
+    /**
+     * The default {@link RequestContextStorage} which stores the {@link RequestContext} in
+     * {@link FastThreadLocal}.
+     */
+    public static final RequestContextStorage
+            threadLocalRequestContextStorage = new ThreadLocalRequestContextStorage();
+
+    private static final RequestContextStorage requestContextStorage;
+
+    static {
+        final List<RequestContextStorageProvider> providers = ImmutableList.copyOf(
+                ServiceLoader.load(RequestContextStorageProvider.class));
+        final String providerFqcn = Flags.requestContextStorageProvider();
+        if (!providers.isEmpty()) {
+
+            RequestContextStorageProvider provider = null;
+            if (providers.size() > 1) {
+                if (providerFqcn.isEmpty()) {
+                    throw new IllegalStateException(
+                            "Found more than one " + RequestContextStorageProvider.class.getSimpleName() +
+                            ". You must specify -Dcom.linecorp.armeria.requestContextStorageProvider=<FQCN>." +
+                            " providers: " + providers);
+                }
+
+                for (RequestContextStorageProvider candidate : providers) {
+                    if (candidate.getClass().getName().equals(providerFqcn)) {
+                        if (provider != null) {
+                            throw new IllegalStateException(
+                                    providerFqcn + " matches more than one " +
+                                    RequestContextStorageProvider.class.getSimpleName() + ". providers: " +
+                                    providers);
+                        } else {
+                            provider = candidate;
+                        }
+                    }
+                }
+                if (provider == null) {
+                    throw new IllegalStateException(
+                            providerFqcn + " does not match any " +
+                            RequestContextStorageProvider.class.getSimpleName() + ". providers: " + providers);
+                }
+            } else {
+                provider = providers.get(0);
```

Comment:
Could you log the type of the provider in this branch, so that what custom provider was chosen? For other branches, `Flag` will log.

Reply:
Added logging after `requestContextStorage` is created.